lib/utils/fs_test.go

```diff
@@ -26,6 +26,80 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestOpenFileSymlinks(t *testing.T) {
+	// symlink structure setup, this will produce the following structure under the temp directory created below:
+	// dir
+	// dir-s -> dir
+	// dir-s-s -> dir-s
+	// dir/file
+	// dir/file-s -> dir/file
+	// circular-s -> circular-s
+	// broken-s -> nonexistent
+	rootDir := t.TempDir()
+
+	dirPath := filepath.Join(rootDir, "dir")
+	err := os.Mkdir(dirPath, 0755)
+	require.NoError(t, err)
+
+	filePath := filepath.Join(dirPath, "file")
```

Comment:
```suggestion
	dirFilePath := filepath.Join(dirPath, "file")
```

So the hierarchy is clearer from the names? Same for others.